backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -60,7 +60,8 @@ export class FeatureFlagService {
     @InjectRepository() private experimentAuditLogRepository: ExperimentAuditLogRepository,
     @InjectDataSource() private dataSource: DataSource,
     public experimentAssignmentService: ExperimentAssignmentService,
-    public segmentService: SegmentService
+    public segmentService: SegmentService,
+    private segmentArray = []
```

Comment:
This service instance is used for multiple http requests and it is possible that it get overwritten by another request. You should not use private variable in the service.